src/tls.rs

```diff
@@ -1,6 +1,234 @@
-//! TLS-related helper types
-use core::ffi::{c_char, CStr};
-use core::fmt::Debug;
+//! Type safe abstraction for esp-tls
+
+use std::{
+    ffi::CStr,
+    fmt::{Debug, Display},
+    io::{self, Read, Write},
+};
+
+pub struct EspTls {
+    raw: *mut esp_idf_sys::esp_tls,
+}
+
+impl EspTls {
+    pub fn new(host: &str, port: i32, cfg: Config) -> Result<Self> {
```

Comment:
* Nit: pass `Config` by ref, i.e. `&Config` as we do everywhere else
* Since we want to decouple this from STD, the return type should be `Result<Self, EspError>`. Do not introduce a custom `Error` type. We use everywhere just `EspError`.
* port should be `u16`